### PR TITLE
test: moved `TopBarMenu` tests from Enzyme to `RTL`

### PR DESCRIPTION
**Pull request by orest-s**
Issue https://github.com/dequelabs/cauldron/issues/1159

This PR switches the `TopBarMenu` tests to React Testing Library.

Validation steps

Run yarn test
See that there are coverage percentages in the table
Verify that the tests are converted properly

### CODE REVIEW

**Comment by anastasialanz** [comment]
I think this is a duplicate of https://github.com/dequelabs/cauldron/pull/1413/files

File: packages/react/src/components/TopBar/TopBarMenu.test.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  expect(
+    screen.getByRole('menuitem', { name: 'option 1' })
+  ).toBeInTheDocument();
+  expect(
+    screen.getByRole('menuitem', { name: 'option 2' })
+  ).toBeInTheDocument();
+});
+
+test('should pass-through props', () => {
+  const restProps = { foo: 'bar' };
```
Comment (anastasialanz) [inline packages/react/src/components/TopBar/TopBarMenu.test.tsx:39]: `foo` isn't a real prop we should be able to pass down if we used the `TopBarMenuProps` interface. I think we'll want to update this example.

**Comment by orest-s** [comment]
@anastasialanz sorry, i didn't notice that someone already opened PR for this component, can i just close my PR?

**Comment by anastasialanz** [comment]
> @anastasialanz sorry, i didn't notice that someone already opened PR for this component, can i just close my PR?

Keep yours open! The other person hasn't responded to feedback in some time.

File: packages/react/src/components/TopBar/TopBarMenu.test.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  });
+
+  expect(menu).toHaveAttribute('aria-expanded', 'false');
+});
+
+test('should close menu with right key', () => {
+  render(
+    <MenuBar>
+      <MenuItem>a</MenuItem>
+      <TopBarMenu {...defaultProps} data-testid="test-top-bar">
```
Comment (anastasialanz) [inline packages/react/src/components/TopBar/TopBarMenu.test.tsx:126]: Can we remove this `data-testid` too since it isn't being used?

**Review by anastasialanz — APPROVED** [review]

**Pull request merged**